app.rb

```diff
@@ -0,0 +1,166 @@
+require_relative 'person'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'rental'
+require_relative 'book'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  # Show menu for users
+  def show_menu
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts "7 - Exit\n\n"
+    gets.chomp
+  end
+
+  # Method that handles user choice
+  def run
+    user_choice = show_menu
+    case user_choice
+    when '1'
+      action_list_books
+    when '2'
+      action_list_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rental
+    else
+      puts 'Thanks for using this app ...'
+      exit
+    end
+  end
+
+  # List all books
+  def list_books
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  # Action list all Books
+  def action_list_books
+    list_books
+    puts 'Press enter to continue ...'
+    gets.chomp
+    run
+  end
+
+  # Permission access for Person (Teacher or Student)
+  def permission(my_char)
+    case my_char
+    when 'N' || 'n'
+      false
+    when 'Y' || 'y'
+      true
+    end
+  end
+
+  # List all People
+  def list_people
+    @people.each_with_index do |person, index|
+      # puts "#{index}) Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      puts "#{index}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  # List all Books
+  def action_list_people
+    list_people
+    puts "\n\nPress any key to continue"
+    gets
+    run
+  end
+
+  # List all rentals for a given person id.
+  def list_rentals(id)
+    selected_person = @people.find { |person| person.id == id }
+    return nil if selected_person.nil?
+
+    selected_person.rentals.each do |rental|
+      puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}"
+    end
+  end
+
+  # Handles creation of a Person (Teacher or Student)
+  def create_person
+    print 'Do you want to create a Student (1) or a Teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+    case person_type
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    end
```

Comment:
- [ ] Kindly add error handling for cases where the user provides neither 1 nor 2 as input. You can add a default case that informs the user to chose between 1 and 2 again.
![image](https://user-images.githubusercontent.com/44057863/175788283-4a7ae52c-5c52-4c71-9828-b6613b4fc0ae.png)

Reply:
I fixed everything except begin, retry and rescue error handling. It will be added tomorrow because I never used it I tried but I need some additional reading.